connect/runtime/src/test/java/org/apache/kafka/connect/runtime/isolation/PluginUtilsTest.java

```diff
@@ -490,6 +504,168 @@ public void testPluginUrlsWithRelativeSymlinkForwards() throws Exception {
         assertUrls(expectedUrls, PluginUtils.pluginUrls(pluginPath));
     }
 
+    @Test
+    public void testNonCollidingAliases() {
+        SortedSet<PluginDesc<SinkConnector>> sinkConnectors = new TreeSet<>();
+        sinkConnectors.add(new PluginDesc<>(MockSinkConnector.class, null, MockSinkConnector.class.getClassLoader()));
+        SortedSet<PluginDesc<SourceConnector>> sourceConnectors = new TreeSet<>();
+        sourceConnectors.add(new PluginDesc<>(MockSourceConnector.class, null, MockSourceConnector.class.getClassLoader()));
+        SortedSet<PluginDesc<Converter>> converters = new TreeSet<>();
+        converters.add(new PluginDesc<>(CollidingConverter.class, null, CollidingConverter.class.getClassLoader()));
+        PluginScanResult result = new PluginScanResult(
+                sinkConnectors,
+                sourceConnectors,
+                converters,
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet(),
+                Collections.emptySortedSet()
+        );
+        Map<String, String> aliases = PluginUtils.computeAliases(result);
+        assertFalse(aliases.containsKey("Mock"));
+        assertEquals(MockSinkConnector.class.getName(), aliases.get("MockSinkConnector"));
+        assertEquals(MockSinkConnector.class.getName(), aliases.get("MockSink"));
+        assertEquals(MockSourceConnector.class.getName(), aliases.get("MockSourceConnector"));
+        assertEquals(MockSourceConnector.class.getName(), aliases.get("MockSource"));
+        assertEquals(CollidingConverter.class.getName(), aliases.get("CollidingConverter"));
+        assertEquals(CollidingConverter.class.getName(), aliases.get("Colliding"));
```

Comment:
We can compare entire `Map` objects here, which gives better coverage and improves assertion error messages:
```suggestion
        Map<String, String> actualAliases = PluginUtils.computeAliases(result);

        Map<String, String> expectedAliases = new HashMap<>();
        expectedAliases.put("MockSinkConnector", MockSinkConnector.class.getName());
        expectedAliases.put("MockSink", MockSinkConnector.class.getName());
        expectedAliases.put("MockSourceConnector", MockSourceConnector.class.getName());
        expectedAliases.put("MockSource", MockSourceConnector.class.getName());
        expectedAliases.put("CollidingConverter", CollidingConverter.class.getName());
        expectedAliases.put("Colliding", CollidingConverter.class.getName());
        assertEquals(expectedAliases, actualAliases);
```